kernel/src/log_segment/tests.rs

```diff
@@ -620,3 +724,439 @@ fn table_changes_fails_with_larger_start_version_than_end() {
     let log_segment_res = LogSegment::for_table_changes(client.as_ref(), log_root, 1, Some(0));
     assert!(log_segment_res.is_err());
 }
+#[test]
+fn test_sidecar_to_filemeta_valid_paths() -> DeltaResult<()> {
+    let log_root = Url::parse("file:///var/_delta_log/")?;
+    let test_cases = [
+        (
+            "example.parquet",
+            "file:///var/_delta_log/_sidecars/example.parquet",
+        ),
+        (
+            "file:///var/_delta_log/_sidecars/example.parquet",
+            "file:///var/_delta_log/_sidecars/example.parquet",
+        ),
+        (
+            "test/test/example.parquet",
+            "file:///var/_delta_log/_sidecars/test/test/example.parquet",
+        ),
+    ];
+
+    for (input_path, expected_url) in test_cases.into_iter() {
+        let sidecar = Sidecar {
+            path: expected_url.to_string(),
+            modification_time: 0,
+            size_in_bytes: 1000,
+            tags: None,
+        };
+
+        let filemeta = LogSegment::sidecar_to_filemeta(&sidecar, &log_root)?;
+        assert_eq!(
+            filemeta.location.as_str(),
+            expected_url,
+            "Mismatch for input path: {}",
+            input_path
+        );
+    }
+    Ok(())
+}
+
+#[test]
+fn test_checkpoint_batch_with_no_sidecars_returns_none() -> DeltaResult<()> {
+    let (_, log_root) = new_in_memory_store();
+    let engine = Arc::new(SyncEngine::new());
+    let checkpoint_batch = add_batch_simple(get_log_schema().clone());
+
+    let mut iter = LogSegment::process_sidecars(
+        engine.get_parquet_handler(),
+        log_root,
+        checkpoint_batch.as_ref(),
+    )?
+    .into_iter()
+    .flatten();
+
+    // Assert no batches are returned
+    assert!(iter.next().is_none());
+
+    Ok(())
+}
+
+#[test]
+fn test_checkpoint_batch_with_sidecars_returns_sidecar_batches() -> DeltaResult<()> {
+    let (store, log_root) = new_in_memory_store();
+    let engine = DefaultEngine::new(
+        store.clone(),
+        Path::from("/"),
+        Arc::new(TokioBackgroundExecutor::new()),
+    );
```

Comment:
Any reason for using DefaultEngine over SyncEngine? Sync avoids having to declare an executor each time. This could save a lot of lines in this PR too.